analyzer/java/com/engflow/bazel/invocation/analyzer/bazelprofile/BazelProfileConstants.java

```diff
@@ -14,15 +14,61 @@
 
 package com.engflow.bazel.invocation.analyzer.bazelprofile;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+
 /** Constants that are specific to Bazel profiles. */
 public class BazelProfileConstants {
   // Thread names: event.name == METADATA_THREAD_NAME && event.args.name == constant below
   public static final String THREAD_CRITICAL_PATH = "Critical Path";
-  public static final String THREAD_GARBAGE_COLLECTOR = "Garbage Collector";
-  public static final String THREAD_MAIN = "Main Thread";
+
+  @VisibleForTesting public static final String THREAD_GARBAGE_COLLECTOR = "Garbage Collector";
+  // See https://github.com/bazelbuild/bazel/commit/a03674e6297ed5f6f740889cba8780d7c4ffe05c
+  private static final String THREAD_GARBAGE_COLLECTOR_OLD = "Service Thread";
+  /**
+   * Returns whether the passed-in thread looks like the garbage collection, both for newer and
+   * older versions of Bazel. See
+   * https://github.com/bazelbuild/bazel/commit/a03674e6297ed5f6f740889cba8780d7c4ffe05c for when
+   * the naming of the garbage collection thread was changed.
+   *
+   * @param thread the thread to check
+   * @return whether the thread looks like it is the garbage collection thread
+   */
+  public static boolean isGarbageCollectorThread(ProfileThread thread) {
+    String name = thread.getName();
+    if (Strings.isNullOrEmpty(name)) {
+      return false;
+    }
+    return name.equals(BazelProfileConstants.THREAD_GARBAGE_COLLECTOR)
+        || name.equals(BazelProfileConstants.THREAD_GARBAGE_COLLECTOR_OLD);
```

Comment:
The naming (`_OLD` as a qualifier of the immediate previous breaking version), and checks (simple OR) are tied to the fact we faced this issue with only two schema versions.

I think that's enough for this use case, or any current Bazel use case since the community usually uses new versions of Bazel and we don't expect to see errors using older Bazel versions (iff hypothetically Bazel deps used different thread names underneath)